.github/workflows/go.yml

```diff
@@ -11,10 +11,12 @@ jobs:
     - name: Set up Go
       uses: actions/setup-go@v4
       with:
-        go-version: "1.21"
+        go-version: "1.23"
     - name: Build
       run: go build -v ./...
     - name: Test
       run: go test -v ./... -coverprofile=coverage.out -covermode=atomic
     - name: Codecov
-      uses: codecov/codecov-action@v3.1.0
+      uses: codecov/codecov-action@v4
+      env:
+        CODECOV_TOKEN: ${{ secrets.CODECOV_TOKEN }}
```

Comment:
Required to not get throttled by CodeCov. Not particularly sensitive (worst case scenario if this leaks is that someone overwrites our code coverage data)